modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/sql/ParserServiceImpl.java

```diff
@@ -55,12 +56,23 @@ public ParsedResult parse(String query) {
 
         assert queryType != null : normalizedQuery;
 
+        AtomicReference<SqlNode> holder = new AtomicReference<>(parsedTree);
+
+        @SuppressWarnings("UnnecessaryLocalVariable")
         ParsedResult result = new ParsedResultImpl(
                 queryType,
                 query,
                 normalizedQuery,
                 parsedStatement.dynamicParamsCount(),
-                () -> IgniteSqlParser.parse(query, StatementParseResult.MODE).statement()
+                () -> {
+                    SqlNode ast = holder.getAndSet(null);
```

Comment:
hacker )

Reply:
Let's add a comment to explain what going on to future readers of the piece of code